### DISCUSSION TITLE
How to recognize language settings?

### DISCUSSION THREAD

**Discussion by tct123** [flet-dev/flet#4304] [General]
How can I read the language settings of my smartphone with flet? @ndonkoHenri

**Comment by ap4499** [comment]
Did you find a way of doing this?

I saw a couple of Flet bits, but I don't think these will return the system locale:
https://flet.dev/docs/reference/types/localeconfiguration
https://flet.dev/docs/reference/types/locale/


I think the below implementation within Flutter would probably give us what we need

```
import 'dart:io';
final String defaultLocale = Platform.localeName; // Returns locale string in the form 'en_US'
```


https://stackoverflow.com/questions/50923906/how-to-get-timezone-language-and-county-id-in-flutter-by-the-location-of-device

**Reply by tct123** [reply]
Would it be possible to add this directly to flet? For example `page.currentlocale()`

**Reply by tct123** [reply]
@ap4499 Where exactly do i need to add:
```dart
import 'dart:io';
final String defaultLocale = Platform.localeName; // Returns locale string in the form 'en_US
```

**Comment by ndonkoHenri** [comment]
There's an opened issue: https://github.com/flet-dev/flet/issues/3010

**Reply by tct123** [reply]
I know. I opened it. But I didn't got an anser there. `locale.getlocale()` is only working on desktop.

**Reply by ndonkoHenri** [reply]
It will be worked on...

**Comment by tct123** [comment]
Could you add that to the roadmap? @ndonkoHenri
